### PR TITLE
Remove 'Server' header from web API template

### LINKED ISSUE #51: Remove 'Server' header from web API template
## Problem
No need to leak information about the web API in the `Server:` HTTP header, so we need to hide this header from every response.

## Proposal
The kestrel options as a `boolean` called `AddServerHeader` to configure this.

## Resources
https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserveroptions.addserverheader?view=aspnetcore-3.0

### PR DESCRIPTION
**Pull request by tomkerkhove**
Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

Closes #51

### CODE REVIEW

File: src/Arcus.Templates.Tests.Integration/Security/v1/SecurityHeaderTests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        private readonly ITestOutputHelper _outputWriter;
+
+        public SecurityHeaderTests(ITestOutputHelper outputWriter)
+        {
+            _configuration = TestConfig.Create();
+            _outputWriter = outputWriter;
+        }
+
+        [Fact]
+        public async Task SecurityHeaders_BasicHttpCall_ServerHeaderNotReturned()
```
Comment (stijnmoreels) [inline src/Arcus.Templates.Tests.Integration/Security/v1/SecurityHeaderTests.cs]: Cool that you use the integration setup! :)

File: src/Arcus.Templates.WebApi/Program.cs [hunk 2]
```diff
@@ -25,12 +25,13 @@ public static IWebHostBuilder CreateWebHostBuilder(string[] args)
             string httpEndpointUrl = "http://+:" + configuration["ARCUS_HTTP_PORT"];
             IWebHostBuilder webHostBuilder =
                 WebHost.CreateDefaultBuilder(args)
-                       .UseConfiguration(configuration)
-                       .UseUrls(httpEndpointUrl)
-                       .ConfigureLogging(logging => logging.AddConsole())
-                       .UseStartup<Startup>();
+                    .ConfigureKestrel(kestrelServerOptions => kestrelServerOptions.AddServerHeader = false)
```
Comment (stijnmoreels) [inline src/Arcus.Templates.WebApi/Program.cs]: Hmm, change in formatting?

File: docs/features/web-api-template.md [hunk 3]
```diff
@@ -23,16 +23,20 @@ When installed, the template can be created with shortname: `arcus-webapi`:
 ## Features
 
 Creates a starter web API project with by default configured:
-* [Exception middleware](https://webapi.arcus-azure.net/features/logging) to log unhandled exceptions thrown during request processing.
-* Content negotiation that only supports `application/json`.
-* Swagger docs generation and UI (only available locally).
-* Provides basic health endpoint with [ASP.NET Core health checks](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-2.2) with [OpenAPI support](https://www.codit.eu/blog/documenting-asp-net-core-health-checks-with-openapi/).
-* Docker building file.
-* Default console logger.
+- [Exception middleware](https://webapi.arcus-azure.net/features/logging) to log unhandled exceptions thrown during request processing.
```
Comment (stijnmoreels) [inline docs/features/web-api-template.md]: Change in formatting, but fine by me.

File: src/Arcus.Templates.WebApi/Program.cs (hunk 2 above)
Comment (tomkerkhove) [reply src/Arcus.Templates.WebApi/Program.cs]: Ah must be my R# settings at work, reverting.

File: src/Arcus.Templates.Tests.Integration/Security/v1/SecurityHeaderTests.cs (hunk 1 above)
Comment (tomkerkhove) [reply src/Arcus.Templates.Tests.Integration/Security/v1/SecurityHeaderTests.cs]: I did a copy paste so I didn't think!

No just kidding, it's very easy to use, thanks!

**Pull request merged**